source/common/event/dispatcher_impl.cc

```diff
@@ -265,9 +269,21 @@ void DispatcherImpl::post(std::function<void()> callback) {
   }
 }
 
+void DispatcherImpl::deleteInDispatcherThread(DispatcherThreadDeletablePtr deletable) {
+  bool do_post;
+  {
+    Thread::LockGuard lock(post_lock_);
+    do_post = post_callbacks_.empty() || deletables_in_dispatcher_thread_.empty();
+    deletables_in_dispatcher_thread_.emplace_back(std::move(deletable));
+  }
+
+  if (do_post) {
+    post_cb_->scheduleCallbackCurrentIteration();
```

Comment:
Please use a different schedulable callback for deletables_in_dispatcher_thread_

Reply:
done